network/src/peer_manager/mod.rs

```diff
@@ -783,7 +783,12 @@ where
     fn add_peer(&mut self, connection: Connection<TSocket>) {
         let conn_meta = connection.metadata.clone();
         let peer_id = conn_meta.remote_peer_id;
-        assert_ne!(self.network_context.peer_id(), peer_id);
+
+        // Make a disconnect if you've connected to yourself
+        if self.network_context.peer_id() == peer_id {
+            self.disconnect(connection);
+            return;
+        }
```

Comment:
we _should_ be guaranteed from the Transport layer that we don't get any self connections. but I guess it doesn't hurt to be defensive. maybe log and add a `debug_assert` and so we can fail fast in testing?

Reply:
i'm imagining future Greg or Philip debugging a self-dial issue and it looks painful 😅

Reply:
We *should*, it's already a real pain to debug these kinds of problems since `peer_id`s are reused across nodes owned by the same account.

I just don't want a place where twins somehow dial each other, and knock each other out, or some weird thing like that...  Or someone maliciously tries to connect with the same peer id as the current node (different key), and knock it out on a public interface.  I'm not sure that can happen, but still

Reply:
agreed, let's not leave hidden traps here for ourselves that can take the node out of commission, but considering that this would result in a silent error as is, let's add that logging!